AICopilot/socket_server.py

```
@@ -2223,34 +2203,31 @@ def _execute_python(self, args: Dict[str, Any]) -> str:
                 'FreeCAD': FreeCAD,
                 'FreeCADGui': FreeCADGui,
                 'doc': FreeCAD.ActiveDocument,
                 'print': lambda *args: FreeCAD.Console.PrintMessage('CODE: ' + ' '.join(str(arg) for arg in args) + '\n')
             }
-            
-            # Execute with detailed logging
+
             FreeCAD.Console.PrintMessage("EXEC: Starting code execution...\n")
-            
             try:
                 exec(code, exec_context)
```

Ruff (0.16.1)

[error] Use of `exec` detected

(S102)